tensorflow/compiler/tf2tensorrt/convert/ops/data_format_vec_permute.cc

```diff
@@ -0,0 +1,184 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifdef GOOGLE_CUDA&& GOOGLE_TENSORRT
+#include "tensorflow/compiler/tf2tensorrt/common/utils.h"
+#include "tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h"
+#include "tensorflow/compiler/tf2tensorrt/convert/op_converter.h"
+#include "tensorflow/compiler/tf2tensorrt/convert/op_converter_registry.h"
+#include "tensorflow/compiler/tf2tensorrt/convert/utils.h"
+#include "tensorflow/core/lib/core/errors.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "third_party/tensorrt/NvInfer.h"
+#include "third_party/tensorrt/NvInferRuntimeCommon.h"
+
+namespace tensorflow {
+namespace tensorrt {
+namespace convert {
+
+class ConvertDataFormatVecPermute
+    : public OpConverterBase<ConvertDataFormatVecPermute> {
+ public:
+  ConvertDataFormatVecPermute(OpConverterParams* params)
+      : OpConverterBase<ConvertDataFormatVecPermute>(params) {}
+
+  struct DataFormatVecPermuteAttributes {
+    string dst_format;
+    string src_format;
+    int spatial_dim_count;
+    int x_dim_count;
+  };
+
+  static constexpr std::array<InputArgSpec, 1> InputSpec() {
+    return {InputArgSpec::Create("x", TrtInputArg::kBoth)};
+  }
+
+  static constexpr std::array<DataType, 1> AllowedDataTypes() {
+    return {DataType::DT_INT32};
+  }
+
+  Status Validate() {
+    const auto& inputs = params_->inputs;
+    const auto& node_def = params_->node_def;
+
+    if (params_->use_implicit_batch) {
+      return errors::Unimplemented("Implicit batch mode not supported, at ",
+                                   node_def.name());
+    }
+
+    x_input_ = inputs.at(0);
+
+    // Check input rank
+    const auto x_dims = x_input_.GetTrtDims();
+    int input_rank = x_dims.nbDims;
+    if (input_rank != 1 && input_rank != 2) {
+      return errors::InvalidArgument(
+          "Input must be a vector or matrix, but got rank ", input_rank,
+          ", at ", node_def.name());
+    }
+
+    // Verify and consume node attributes
+    StatusOr<string> dst_format = GetAttrValue<string>("dst_format");
+    StatusOr<string> src_format = GetAttrValue<string>("src_format");
+    TRT_ENSURE_OK(dst_format);
+    TRT_ENSURE_OK(src_format);
+    attrs_.dst_format = *dst_format;
+    attrs_.src_format = *src_format;
+
+    // Check input dims
+    const int full_dim_count = attrs_.src_format.size();
+    const int spatial_dim_count = full_dim_count - 2;
```

Comment:
Can we add this document here:

//Spatial dimensions are the dimensions besides NC, and here we assume NC always appear in the format string.